src/main/scala/com.snowplowanalytics.snowplow.analytics.scalasdk/SnowplowEvent.scala

```diff
@@ -0,0 +1,50 @@
+package com.snowplowanalytics.snowplow.analytics.scalasdk
+
+// java
+import java.util.UUID
+
+// circe
+import io.circe.syntax._
+import io.circe.{Encoder, Json, JsonObject}
+
+// This library
+import com.snowplowanalytics.iglu.core.SelfDescribingData
+
+trait SnowplowEvent {
+  def event_id: UUID
+}
+
+object SnowplowEvent {
+
+  case class UnstructEvent(data: Option[SelfDescribingData[Json]]) extends AnyVal
+
+  case class Contexts(data: List[SelfDescribingData[Json]]) extends AnyVal
+
+  implicit final val contextsCirceEncoder: io.circe.Encoder[Contexts] =
+    Encoder.instance { contexts: Contexts =>
+      if (contexts.data.isEmpty) List[String]().asJson
+      else JsonObject(
+        ("schema", "iglu:com.snowplowanalytics.snowplow/contexts/jsonschema/1-0-0".asJson),
```

Comment:
This Iglu URI is very important, it is worth to have it as a constant somewhere and switch to structured `SchemaKey` rather than lossy `String`.

Reply:
Also, there's `SelfDescribingData[Json]` container in Iglu Core, I encourage to use it everywhere as much as possible. It also has `normalize` method (as well as `asJson` if codecs are imported), which will be much more concise than `JsonOject().asJson`